akka-http-cors/src/main/scala/ch/megard/akka/http/cors/scaladsl/settings/CorsSettings.scala

```diff
@@ -119,6 +122,37 @@ abstract class CorsSettings extends javadsl.settings.CorsSettings {
 
 object CorsSettings {
 
+  private val prefix = "akka.http.cors"
+
+  def apply(config: Config) = fromSubConfig(config.getConfig(prefix))
+
+  def fromSubConfig(config: Config) = CorsSettings.Default(
+    allowGenericHttpRequests = config.getBoolean("allow-generic-http-requests"),
+    allowCredentials = config.getBoolean("allow-credentials"),
+    allowedOrigins = config.getStringList("allowed-origins").asScala.toList match {
+      case List("*") => HttpOriginRange.*
+      case origins => HttpOriginRange(origins.map(HttpOrigin(_)): _*)
+    },
+    allowedHeaders = config.getStringList("allowed-headers").asScala.toList match {
+      case List("*") => HttpHeaderRange.*
+      case headers => HttpHeaderRange(headers: _*)
+    },
+    allowedMethods = config.getStringList("allowed-methods").asScala.toList.map(method =>
+      HttpMethods.getForKey(method) match {
+        case Some(httpMethod) => httpMethod
+        case None => HttpMethod.custom(method)
+      }
+    ),
+    exposedHeaders = config.getStringList("exposed-headers").asScala.toList,
+    maxAge = Try(config.getDuration("max-age", TimeUnit.SECONDS)) match {
```

Comment:
As you found out, default numbers are interpreted as milliseconds. That _could_ work but I don't think it's a good default since the CORS header takes seconds.
You can pattern march on `config.getValue(???).valueType` to know the type before actually reading it. Also the `null` should not be handled by throwing an exception. Use the `getIsNull()` method instead? Feel free to create local helper functions to parse these values. For example, a `parseSeconds()` function like this:
```
def parseSeconds(path: String): Option[Long] = {
  if (config.getIsNull(path)) {
    None
  } else {
    config.getValue(path).valueType match {
      ...
    }
  }
}
```

It's not very clean but with a Java Api it's the best we can do :D